lib/graphql/schema/resolver.rb

```diff
@@ -91,6 +91,15 @@ def resolve(**args)
       def before_prepare(**args)
       end
 
+      # Called after arguments are loaded, but before resolving.
+      #
+      # Override it to check everything before calling the mutation.
+      # @param args [Hash] The input arguments
+      # @raise [GraphQL::ExecutionError] To add an error to the response
+      # @raise [GraphQL::UnauthorizedError] To signal an authorization failure
+      def authorized?(**args)
```

Comment:
The name `authorized?` might imply that the method always returns a boolean? But the default implementation returns `nil`. Do you think it would make sense to use `!` similar to methods in Rails like [`last!`](https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-last-21) or [`find_by!`](https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find_by-21) that raise exceptions?

Reply:
oh yeah, maybe it should be updated to handle a true/false return value, or renamed like you said

Reply:
Maybe accepting `false` would solve #1737  too, you could do something like 

```
# Don't continue execution, instead, use this hash as the mutation's return value
return false, { errors: validation_errors }
```

Reply:
updated in aa8b5f8a